tests/input-test.js

```diff
@@ -170,29 +170,40 @@ describe('Input', () => {
         inputNode.value = 'aaaa';
         input.setCaretPos(4);
         TestUtils.Simulate.change(inputNode);
-        expect(inputNode.value).toEqual('aaaa');
+        expect(inputNode.value).toEqual('aaaa ');
         expect(input.getCaretPos()).toEqual(4);
 
-        inputNode.value = 'aaaaa';
-        input.setCaretPos(5);
-        TestUtils.Simulate.change(inputNode);
-        expect(inputNode.value).toEqual('aaaa a');
-        expect(input.getCaretPos()).toEqual(6);
+        ReactDOM.unmountComponentAtNode(container);
+    }));
 
-        inputNode.value = 'aaaa afgh ijkl mnop';
-        input.setCaretPos(19);
-        TestUtils.Simulate.change(inputNode);
-        expect(inputNode.value).toEqual('aaaa afgh ijkl mnop');
-        expect(input.getCaretPos()).toEqual(19);
+    it('onChange input without mask 2', createInput(
+      <Input mask="**** **** **** ****" maskChar={null} />, (input) => {
+          var inputNode = ReactDOM.findDOMNode(input);
 
-        inputNode.value = 'aaaa afgh ijkl mnopq';
-        input.setCaretPos(20);
-        TestUtils.Simulate.change(inputNode);
-        expect(inputNode.value).toEqual('aaaa afgh ijkl mnop');
-        expect(input.getCaretPos()).toEqual(19);
+          inputNode.focus();
+          TestUtils.Simulate.focus(inputNode);
 
-        ReactDOM.unmountComponentAtNode(container);
-    }));
+          input.setCaretPos(0);
+          inputNode.value = 'aaaaa';
+          input.setCaretPos(5);
+          TestUtils.Simulate.change(inputNode);
+          expect(inputNode.value).toEqual('aaaa a');
+          expect(input.getCaretPos()).toEqual(6);
```

Comment:
The reason I've split one test suit into 2 is this line of code. It throws an error `5 is not equal 6` ONLY if this test follows the test on lines 170-174.

It might be a bug in PhantomJS browser(which I've used to test this) or maybe in use of `raf` for setting selection. Either way I haven't found the reason for this. `start` selection was 5 in that case. It works fine in real browser, though, even without this workaround.